### PR TITLE
[Chore](column) support all column method on column_fixed_object and remove some method about IColumn

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
support all column method on column_fixed_object and remove some method about IColumn
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

File: be/src/vec/columns/column_fixed_length_object.h [hunk 1]
```diff
@@ -159,60 +161,65 @@ class ColumnFixedLengthObject final : public COWHelper<IColumn, ColumnFixedLengt
         memset(&_data[old_size * _item_size], 0, _item_size);
     }
 
-    void pop_back(size_t n) override { LOG(FATAL) << "pop_back not supported"; }
+    void pop_back(size_t n) override { resize(_item_count - n); }
```
Comment (HappenLee) [inline be/src/vec/columns/column_fixed_length_object.h:164]: DCHECK or check `n <= _item_count`

File: be/src/vec/columns/column_nullable.h [hunk 2]
```diff
@@ -353,23 +349,41 @@ class ColumnNullable final : public COWHelper<IColumn, ColumnNullable> {
...
-
     MutableColumnPtr convert_to_predicate_column_if_dictionary() override {
         nested_column = get_nested_column().convert_to_predicate_column_if_dictionary();
         return get_ptr();
     }
 
     double get_ratio_of_default_rows(double sample_ratio) const override {
-        return get_ratio_of_default_rows_impl<ColumnNullable>(sample_ratio);
+        if (sample_ratio <= 0.0 || sample_ratio > 1.0) {
+            LOG(FATAL) << "Value of 'sample_ratio' must be in interval (0.0; 1.0], but got: "
```
Comment (HappenLee) [inline be/src/vec/columns/column_nullable.h:362]: throw Exception ?

File: be/src/vec/columns/column_nullable.h [hunk 3]
```diff
@@ -353,23 +349,41 @@ class ColumnNullable final : public COWHelper<IColumn, ColumnNullable> {
...
+                       << sample_ratio;
+        }
+        static constexpr auto max_number_of_rows_for_full_search = 1000;
```
Comment (HappenLee) [inline be/src/vec/columns/column_nullable.h:365]: MAX_NUMBER

File: be/src/vec/columns/column_nullable.h [hunk 4]
```diff
@@ -353,23 +349,41 @@ class ColumnNullable final : public COWHelper<IColumn, ColumnNullable> {
...
-        LOG(FATAL) << "should not call the method in column nullable";
-        __builtin_unreachable();
-    }
-
     MutableColumnPtr convert_to_predicate_column_if_dictionary() override {
         nested_column = get_nested_column().convert_to_predicate_column_if_dictionary();
         return get_ptr();
     }
 
     double get_ratio_of_default_rows(double sample_ratio) const override {
```
Comment (HappenLee) [inline be/src/vec/columns/column_nullable.h:360]: should we really need the func? seems unless in doris code

Comment (BiteTheDDDDt) [reply be/src/vec/columns/column_nullable.h:360]: used on ColumnObject

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**